### PR TITLE
Release v0.7.0

### PR DESCRIPTION
**Pull request by ahms5**
0.7.0 (2024-09-20)
------------------
* Features

  * Signals and TimeData objects can now have complex time data. This is specified by the `complex` flag. In the case of Signal objects, this results in both sided spectra (PR #504, #396, #462, #464, #505, #510, #513, #547, #646, #655, #668, #704)
    * Functions in the dsp module can now handle complex valued time data or raise an error (PR #459, #460, #461, #525, #546, #566, #567, #573, #592, #594, #598, #601, #677)
    * All plot functions can now handle complex valued time data and both sided spectra via the options `side` and `mode`  (PR #615, #652, #656, #671, #672, #702)
  * Added TransmissionMatrix Class, which is a convenient representations for two-ports / quadrupoles. These can represent systems from various fields, e.g. electrical circuits, mechanical vibration, (acoustic) transmission lines (PR #636).
  * Added frequency domain sweep synthesis (PR #199)
  * Added spectral soft limiting (PR #325)
  * Added arithmetic operations for Coordinate class (PR #680)
  * Added transpose method to audio classes (PR #481)
  * Added allpass filters (PR #571)
  * Added `cdim` property to audio classes (PR #619)

* Enhancements

  * The sampling_rate property of Signal Objects is enforced to be a scalar. (PR #534)
ahms5 marked this conversation as resolved.
Hide resolved
  * The ticks in phase plots adjust better to the range of the data shown in the plot (PR #559)
  * Replaced scipy's legacy spectrogram with new short-time Fourier transform implementation (PR #682)
  * Deprecated Python 3.8 (PR #683)

* Bugfixes

  * Removed phase for FFT normalizations 'power' and 'psd' (PR #557)
  * Fix unit string in group delay plot (PR #703)
  * Rename and deprecate shelve/shelf, shelve_type/shelf_type due to a typo in the function names (PR #643)

* Documentation

  * Improve docstring for pyfar.signals.exponential_sweep_time (PR #514)
  * Show paths of classes as imported in pyfar (PR #684)

* Automation and testing

  * replace flake8 by ruff (PR #526, #611, #686, #687, #688, #689, #690, #691, #702)
  * remove environment files which are no longer required (PR #626)
  * Harmonize class documentation (PR #631)
  * wait for test_deprecation_warnings before release (PR #696)

### CODE REVIEW

**Comment by ahms5** [comment]
ok maybe we need to update the data^^

File: HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,46 @@
...
+0.7.0 (2024-09-20)
+------------------
+* Features
+
+  * Signals and TimeData objects can now have complex time data. This is specified by the `complex` flag. In the case of Signal objects, this results in both sided spectra (PR #504, #396, #462, #464, #505, #510, #513, #547, #646, #655, #668, #704)
+    * Functions in the dsp module can now handle complex valued time data or raise an error (PR #459, #460, #461, #525, #546, #566, #567, #573, #592, #594, #598, #601, #677)
+    * All plot functions can now handle complex valued time data and both sided spectra via the options `side` and `mode`  (PR #615, #652, #656, #671, #672, #702)
+  * Added TransmissionMatrix Class (PR #636)
+  * Added frequency domain sweep synthesis (PR #199)
+  * Added spectral soft limiting(PR #325)
```
Comment (ahms5) [inline HISTORY.rst:14]:
```suggestion
  * Added spectral soft limiting (PR #325)
```

File: HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,46 @@
...
+  * Add arithmetic operations for Coordinate class (PR #680)
```
Comment (ahms5) [inline HISTORY.rst:15]:
```suggestion
  * Added arithmetic operations for Coordinate class (PR #680)
```

File: HISTORY.rst [hunk 3]
```diff
@@ -2,6 +2,46 @@
...
+  * Added transpose method to audio classes (PR #481)
+  * Added allpass filters (PR #571)
+  * Added `cdim`` property to audio classes (PR #619)
+
+* Enhancements
+
+  * The sampling_rate property of Signal Objects is enforced to be a scalar. (PR #534)
```
Comment (ahms5) [inline HISTORY.rst:22]: isnt sth like sampling_rate=[1000] and sampling_rate=np.array([1000]) aslo allowed?

File: HISTORY.rst (hunk 1 above)
Comment (f-brinkmann) [reply HISTORY.rst]: fixed

File: HISTORY.rst (hunk 2 above)
Comment (f-brinkmann) [reply HISTORY.rst]: fixed

File: HISTORY.rst (hunk 3 above)
Comment (f-brinkmann) [reply HISTORY.rst:22]: I would say its allowed but not desired. And during initialization `signal = pf.Signal([1, 2, 3], np.array([1]))` `signal.sampling_rate` is caseted to `np.int`, which I think is the important enhancement.

Comment (ahms5) [reply HISTORY.rst:22]: Yes, that makes sense. Thank you!

**Comment by ahms5** [comment]
I would approve, but I cant :D

File: HISTORY.rst [hunk 4]
```diff
@@ -2,6 +2,46 @@
...
+  * Signals and TimeData objects can now have complex time data. This is specified by the `complex` flag. In the case of Signal objects, this results in both sided spectra (PR #504, #396, #462, #464, #505, #510, #513, #547, #646, #655, #668, #704)
+    * Functions in the dsp module can now handle complex valued time data or raise an error (PR #459, #460, #461, #525, #546, #566, #567, #573, #592, #594, #598, #601, #677)
+    * All plot functions can now handle complex valued time data and both sided spectra via the options `side` and `mode`  (PR #615, #652, #656, #671, #672, #702)
+  * Added TransmissionMatrix Class (PR #636)
+  * Added frequency domain sweep synthesis (PR #199)
+  * Added spectral soft limiting (PR #325)
+  * Added arithmetic operations for Coordinate class (PR #680)
+  * Added transpose method to audio classes (PR #481)
+  * Added allpass filters (PR #571)
+  * Added `cdim`` property to audio classes (PR #619)
```
Comment (ahms5) [inline HISTORY.rst:18]:
```suggestion
  * Added `cdim` property to audio classes (PR #619)
```

File: HISTORY.rst [hunk 5]
```diff
@@ -2,6 +2,46 @@
...
 =======
 
+0.7.0 (2024-09-20)
+------------------
+* Features
+
+  * Signals and TimeData objects can now have complex time data. This is specified by the `complex` flag. In the case of Signal objects, this results in both sided spectra (PR #504, #396, #462, #464, #505, #510, #513, #547, #646, #655, #668, #704)
+    * Functions in the dsp module can now handle complex valued time data or raise an error (PR #459, #460, #461, #525, #546, #566, #567, #573, #592, #594, #598, #601, #677)
+    * All plot functions can now handle complex valued time data and both sided spectra via the options `side` and `mode`  (PR #615, #652, #656, #671, #672, #702)
+  * Added TransmissionMatrix Class (PR #636)
```
Comment (ph-schaefer) [inline HISTORY.rst:12]:
```suggestion
  * Added TransmissionMatrix Class (PR #636). Transmission matrices are a convenient representations for two-ports / quadrupoles. These can represent systems from various fields, e.g. electrical circuits, mechanical vibration, (acoustic) transmission lines.
```

Comment (ahms5) [reply HISTORY.rst:12]:
```suggestion
  * Added TransmissionMatrix Class, which is a convenient representations for two-ports / quadrupoles. These can represent systems from various fields, e.g. electrical circuits, mechanical vibration, (acoustic) transmission lines (PR #636).
```
maybe like this?

File: HISTORY.rst [hunk 6]
```diff
@@ -2,6 +2,46 @@
...
+  * Rename and deprecate shelve/shelf, shelve_type/shelf_type due to a typo in the function names (PR #643)
+
+* Documentation
+
+  * Improve docstring for pyfar.signals.exponential_sweep_time (PR #514)
+  * Show paths of classes as imported in pyfar (PR #684)
+
+* Automation and testing
+
+  * replace flake8 by ruff (PR #526, #611, #686, #687, #688, #689, #690, #691, #702)
```
Comment (ahms5) [inline HISTORY.rst:40]:
```suggestion
  * replace flake8 by ruff (PR #526, #611, #686, #687, #688, #689, #690, #691, #702, #699, #697)
```

**Pull request merged**